packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -45,6 +55,10 @@ const propTypes = {
    * A callback function triggered when the input or calendar button loses focus.
    */
   onBlur: PropTypes.func,
+  /**
+   * A callback function triggered when the calendar button receives focus.
+   */
+  onButtonFocus: PropTypes.func,
```

Comment:
nit : can these be named as `onCalendarButtonFocus` OR `onDatePickerButtonFocus` to distinguish between input `onFocus`...? and also `onFocus` to `onInputFocus`

Reply:
I have kept the props naming convention exactly the same as the previous implementation to keep majority of the changes localised in the "DateInput.jsx" file.

Reply:
Can you add the param(s) that onButtonFocus will be called with?

Reply:
`onButtonFocus` just takes the event object as a param, there are no extra parameters being passed.
Do i still need to add the param documentation?
If yes, then ill go ahead and add it for all the other callbacks as well callbacks.